File: src/main/java/com/king/spark/sql/DataFrameCreate.java
package com.king.spark.sql;


import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SparkSession;

/**
 * 使用json文件创建DataFrame
 *
 * @author king
 */
public class DataFrameCreate {

    public static void main(String[] args) {
        SparkSession spark = SparkSession
                .builder()
                .master("local")
                .appName("DataFrameCreate")
                .getOrCreate();

        Dataset df = spark.read().json("E:\\Git\\spark-study-scala\\file\\data\\json\\student.json");

        df.show();
    }

}

File: src/main/scala/com/king/spark/sql/DataFrameCreate.scala
package com.king.spark.sql

import org.apache.spark.sql.SparkSession

/**
  * @author king
  */
object DataFrameCreate {

  def main(args: Array[String]) {
    val spark = SparkSession
      .builder()
      .master("local")
      .appName("DataFrameOperation")
      .getOrCreate()
    //hdfs目录
    //val df = spark.read.json("hdfs://spark1:9000/students.json")
    val df = spark.read.json("E:\\Git\\spark-study-scala\\file\\data\\json\\student.json")

    df.show()
  }

}